src/app/recipes/recipe-from/recipe-from.component.html

```diff
@@ -0,0 +1,291 @@
+<h2 class="mb-3">Recipe</h2>
+<form
+	[formGroup]="recipeForm"
+	#ngForm="ngForm"
+	(ngSubmit)="handleSubmit()"
+	*ngIf="!loading">
+	<div class="row">
+		<div class="col-12 col-md-9 col-xl-8">
+			<div class="title">
+				<label for="title" class="form-label fs-5 fw-bold">{{
+					'Recipes.Title' | translate
+				}}</label>
+				<input
+					type="text"
+					formControlName="title"
+					[placeholder]="'Recipes.Title' | translate"
+					class="form-control"
+					[class.is-invalid]="
+						(title?.touched || ngForm.submitted) && !!title?.errors
+					"
+					id="title" />
+				<div
+					class="erros"
+					*ngIf="(title?.touched || ngForm.submitted) && !!title?.errors">
+					<div class="text-danger" *ngIf="title?.errors?.['required']">
+						<small>{{
+							'App.Validation.Required' | translate: { field: 'Title' }
+						}}</small>
+					</div>
+					<div
+						class="text-danger"
+						*ngIf="!!title?.errors?.['minlength'] || !!title?.errors?.['maxLength']">
+						<small>
+							{{
+								'App.Validation.Length'
+									| translate: { field: 'Title', min: 10, max: 100 }
+							}}
+						</small>
+					</div>
+				</div>
+			</div>
+
+			<div class="mt-4 imageUrl">
+				<label for="imageUrl" class="form-label fs-5 fw-bold">{{
+					'Recipes.ImageUrl' | translate
+				}}</label>
+				<input
+					type="text"
+					formControlName="imageUrl"
+					[placeholder]="'Recipes.ImageUrl' | translate"
+					class="form-control"
+					[class.is-invalid]="
+						(imageUrl?.touched || ngForm.submitted) && !!imageUrl?.errors
+					"
+					id="imageUrl" />
+				<div
+					class="erros"
+					*ngIf="(imageUrl?.touched || ngForm.submitted) && !!imageUrl?.errors">
+					<div class="text-danger" *ngIf="imageUrl?.errors?.['required']">
+						<small>{{
+							'App.Validation.Required' | translate: { field: 'Image url' }
+						}}</small>
+					</div>
+					<div
+						class="text-danger"
+						*ngIf="!!imageUrl?.errors?.['minlength'] || !!imageUrl?.errors?.['maxLength']">
+						<small>
+							{{
+								'App.Validation.Length'
+									| translate: { field: 'Image Url', min: 10, max: 100 }
+							}}
+						</small>
```

Comment:
think about creating a component for errors handling with
AbstractControl as an Input() and list all possible errors in this component

Reply:
I create form error component